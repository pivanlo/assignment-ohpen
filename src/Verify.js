import React from 'react';
import ReactDOM from 'react-dom';
import { Link, Redirect } from "react-router-dom";
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import { Auth } from "aws-amplify";

class Verify extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      verificationCode: "",
      formError: ""
    };

    this.handleChange = this.handleChange.bind(this);
    this.clearFormError = this.clearFormError.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ verificationCode: event.target.value });
  }

  clearFormError(event) {
    this.setState({ formError: null });
  }

  async handleSubmit(event) {
    event.preventDefault();

    const verificationCode = this.state.verificationCode;

    if (!verificationCode) {
      this.setState({ formError: 'The code cannot be empty' });
    }

    try {
      const response = await Auth.confirmSignUp(
        this.props.auth.username,
        verificationCode
      );
      this.props.history.push("/profile");
    } catch (error) {
      if (error.code === 'CodeMismatchException') {
        this.setState({ formError: 'The code is invalid.' });
      } else if (error.code === 'LimitExceededException'){
        this.setState({ formError: 'Attempt limit exceeded, please try again after later.' });
      } else {
        console.log(error);
      }
    }
  }

  render() {
    if (!this.props.auth.username) {
      return <Redirect to='/sign-in' />;
    }

    return (
      <div className="formContainer">
        <Form className="form-signin" onSubmit={this.handleSubmit}>
          <h1>Verification</h1>

          <Form.Group controlId="formGroupMobileNumber">
            <div className="mb-2">Please enter the verification code that was sent to your email.</div>
            <Form.Control type="number" value={this.state.verificationCode}
              onChange={this.handleChange}
              onFocus={this.clearFormError}
            />
          </Form.Group>

          <Button type="submit" variant="dark" block>Verify</Button>

          {this.state.formError &&
            <div className="mt-3">
              <b>{this.state.formError}</b>
            </div>
          }
        </Form>
      </div>
    );
  }
}

export default Verify;
