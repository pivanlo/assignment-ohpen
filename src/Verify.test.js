import React from 'react';
import Verify from "./Verify.js";
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const auth = {
    username: 'john',
    user: {
      attributes: {
        email: 'john.doe@gmail.com'
      }
    },
    isAuthenticated: false,
    isManager: () => false
  };

  const tree = renderer
    .create(
      <BrowserRouter>
        <Verify auth={auth} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
