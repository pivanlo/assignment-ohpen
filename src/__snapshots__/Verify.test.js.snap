// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="formContainer"
>
  <form
    className="form-signin"
    onSubmit={[Function]}
  >
    <h1>
      Verification
    </h1>
    <div
      className="form-group"
    >
      <div
        className="mb-2"
      >
        Please enter the verification code that was sent to your email.
      </div>
      <input
        className="form-control"
        id="formGroupMobileNumber"
        onChange={[Function]}
        onFocus={[Function]}
        type="number"
        value=""
      />
    </div>
    <button
      className="btn btn-dark btn-block"
      disabled={false}
      type="submit"
    >
      Verify
    </button>
    
  </form>
</div>
`;
